pom.xml

```diff
@@ -1124,6 +1124,38 @@
         <osgi.snapshot.qualifier>qualifier</osgi.snapshot.qualifier>
       </properties>
     </profile>
+    <profile>
+      <id>java9-specific-config</id>
+      <activation>
+        <jdk>[1.9,)</jdk>
+      </activation>
+      <properties>
+        <!-- There is no public ECJ release (yet) which supports Java 9, so fallback to the native compiler -->
+        <drools.dialect.java.compiler>NATIVE</drools.dialect.java.compiler>
```

Comment:
I was under the impression that it does, but obviously it is not that easy. I've added specific configs for both surefire and failsafe.